app/code/Magento/Sales/Model/Order.php

```diff
@@ -1721,7 +1731,21 @@ public function getOrderCurrency()
      */
     public function formatPrice($price, $addBrackets = false)
     {
-        return $this->formatPricePrecision($price, 2, $addBrackets);
+        return $this->formatPricePrecision($price, $this->getPricePrecision(), $addBrackets);
+    }
+
+    /**
+     * @return int
+     */
+    protected function getPricePrecision()
```

Comment:
Please change visibility to private for backward compatibility purpose

Reply:
@sivaschenko done